File: GridView/VirtualRadGridView/VirtualRadGridViewVB/VirtualRadGridViewSortingFiltering.Core/ExpressionParser.vb

Imports System.Collections.Generic
Imports System.Reflection
Imports System.Text
Imports Telerik.Data.Expressions

Namespace VirtualRadGridViewSortingFiltering.Core
    Public Class ExpressionParser
        Private Delegate Function ParseDelegate(expression As String, caseSensitive As Boolean) As ExpressionNode
        Private Shared m_parse As ParseDelegate

        Shared Sub New()
            For Each telerikAssembly As Assembly In AppDomain.CurrentDomain.GetAssemblies()
                Dim expressionParserType As Type = telerikAssembly.[GetType]("Telerik.Data.Expressions.ExpressionParser")
                If expressionParserType IsNot Nothing Then
                    Dim parseMethod As MethodInfo = expressionParserType.GetMethod("Parse")
                    parse = DirectCast([Delegate].CreateDelegate(GetType(ParseDelegate), parseMethod), ParseDelegate)
                    Return
                End If
            Next
        End Sub

        Public Shared Function Parse(expression As String, caseSensitive As Boolean) As ExpressionNode
            Return m_parse(expression, caseSensitive)
        End Function
    End Class
End Namespace

'=======================================================
'Service provided by Telerik (www.telerik.com)
'Conversion powered by NRefactory.
'Twitter: @telerik
'Facebook: facebook.com/telerik
'=======================================================

File: GridView/VirtualRadGridView/VirtualRadGridViewSortingFilteringAfterQ3_2015/VirtualRadGridViewSortingFilteringAfterQ3_2015CS/VirtualRadGridViewSortingFiltering.Core/ExpressionParser.cs
using System;
using System.Collections.Generic;
using System.Reflection;
using System.Text;
using Telerik.Data.Expressions;

namespace VirtualRadGridViewSortingFiltering.Core
{
    public class ExpressionParser
    {
        private delegate ExpressionNode ParseDelegate(string expression, bool caseSensitive);
        private static ParseDelegate parse;

        static ExpressionParser()
        {
            foreach (Assembly telerikAssembly in AppDomain.CurrentDomain.GetAssemblies())
            {
                Type expressionParserType = telerikAssembly.GetType("Telerik.Data.Expressions.ExpressionParser");
                if (expressionParserType != null)
                {
                    MethodInfo parseMethod = expressionParserType.GetMethod("Parse");
                    parse = (ParseDelegate)Delegate.CreateDelegate(typeof(ParseDelegate), parseMethod);
                    return;
                }
            }
        }

        public static ExpressionNode Parse(string expression, bool caseSensitive)
        {
            return parse(expression, caseSensitive);
        }
    }
}
